src/dcmp/dcmp.c

```diff
@@ -646,6 +656,76 @@ do {                                                                         \
     }                                                                        \
 } while(0)
 
+static void dcmp_compare_acl(
+    const char *key,
+    mfu_flist src_list,
+    uint64_t src_index,
+    mfu_flist dst_list,
+    uint64_t dst_index,
+    strmap* src_map,
+    strmap* dst_map,
+    int *diff)
+{
+    void *src_val, *dst_val;
+    ssize_t src_size, dst_size;
+    bool is_same = true;
+
+#if DCOPY_USE_XATTRS
+    src_val = mfu_flist_file_get_acl(src_list, src_index, &src_size,
+                                     "system.posix_acl_access");
+    dst_val = mfu_flist_file_get_acl(dst_list, dst_index, &dst_size,
+                                     "system.posix_acl_access");
+
+    if (src_size == dst_size) {
+        if (src_size > 0) {
+            if (memcmp(src_val, dst_val, src_size)) {
+                is_same = false;
+                goto out;
+            }
+        }
+    } else {
+        is_same = false;
+        goto out;
+    }
+
+    mfu_filetype type = mfu_flist_file_get_type(src_list, src_index);
+    if (type == MFU_TYPE_DIR) {
+        mfu_free(&src_val);
+        mfu_free(&dst_val);
+
+        src_val = mfu_flist_file_get_acl(src_list, src_index, &src_size,
+                                         "system.posix_acl_default");
+        dst_val = mfu_flist_file_get_acl(dst_list, dst_index, &dst_size,
+                                         "system.posix_acl_default");
+
+        if (src_size == dst_size) {
+            if (src_size > 0) {
+                if (memcmp(src_val, dst_val, src_size)) {
+                    is_same = false;
+                    goto out;
+                }
+            }
+        } else {
+            is_same = false;
+            goto out;
+        }
+    }
+
+out:
+    mfu_free(&src_val);
+    mfu_free(&dst_val);
+
+#endif
+    if (is_same) {
+        dcmp_strmap_item_update(src_map, key, DCMPF_ACL, DCMPS_COMMON);
+        dcmp_strmap_item_update(dst_map, key, DCMPF_ACL, DCMPS_COMMON);
+    } else{
```

Comment:
@KnightKu This looks good. Just add a space after the "else" here, and I will merge. Thanks!